provider/ec2/internal/ec2instancetypes/process_cost_data.go

```diff
@@ -247,13 +244,35 @@ func process(in io.Reader) (map[string][]instanceType, metadata, error) {
 		if !ok {
 			return nil, metadata{}, errors.Errorf("unknown location %q", productInfo.Location)
 		}
+		if !supported(region, instanceType.Name) {
+			continue
+		}
+
 		regionInstanceTypes := instanceTypes[region]
 		regionInstanceTypes = append(regionInstanceTypes, instanceType)
 		instanceTypes[region] = regionInstanceTypes
 	}
 	return instanceTypes, meta, nil
 }
 
+// It appears that instances sometimes show up in the offer list which aren't
+// available in actual regions. e.g. m3.medium is in the offer list for
+// ap-northeast-2, but attempting to launch one returns an unsupported error.
+// See: https://bugs.launchpad.net/juju/+bug/1663047
+func supported(region, instanceType string) bool {
+	switch region {
+	case "ap-northeast-2":
```

Comment:
Correct. They are listed with prices in the offer list, but if you try to launch one, it fails with an unsupported error.